### PR TITLE
Initial commit of ocp_client_install role

### PR DESCRIPTION
**Pull request by tehsmyers**

### CODE REVIEW

**Comment by tehsmyers** [comment]
Jenkinsy CI for this role is coming soon, provided this entire approach isn't rejected by my OASISy teammates. :)

File: templates/ocp_install_default_config.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+compute:
+{{ ocp_install_compute | to_nice_yaml(indent=2) | trim | indent(width=2, first=True) }}
+controlPlane:
+{{ ocp_install_control_plane | to_nice_yaml(indent=2) | trim | indent(width=2, first=True) }}
+metadata:
+{{ ocp_install_metadata | to_nice_yaml(indent=2) | trim | indent(width=2, first=True) }}
+networking:
+{{ ocp_install_networking | to_nice_yaml(indent=2) | trim | indent(width=2, first=True) }}
+platform:
+{{ ocp_install_control_plane | to_nice_yaml(indent=2) | trim | indent(width=2, first=True) }}
```
Comment (tehsmyers) [inline templates/ocp_install_default_config.yaml]:
```suggestion
{{ ocp_install_platform | to_nice_yaml(indent=2) | trim | indent(width=2, first=True) }}
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+### `install-config.yaml` Generation
+
+This role can generate an `install-config.yaml`, if requested, placing it in the
+provided `ocp_install_config_dir`.
+
+* `ocp_install_generate_config` - Default: `false`. Whether or not to generate an
+  `install-config.yaml` in the `ocp_install_config_dir`. Forced to `true` if
+  `ocp_install_create_cluster` is `true`.
+* `ocp_install_config_dir` - Default: `~/ocp_install`. Direcotry in which to download
```
Comment (robled) [inline README.md]: Typo - Directory

File: README.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+      clusterNetwork:
+        - cidr: 10.128.0.0/14
+          hostPrefix: 23
+      networkType: OpenShiftSDN
+      serviceNetwork:
+        - 172.30.0.0./16
+    ocp_install_platform:
+      none: {}
+    ocp_install_pull_secrets:
+      super: secret
```
Comment (robled) [inline README.md]: What does `super` do here?  I'm uncertain as to how this expands to the auth blob that's generated by the pull secrets role.

**Comment by tehsmyers** [comment]
I've been writing an actual deployment playbook with this role and found things I'd like to fix in the initial implementation, so that's in progress now.

**Comment by tehsmyers** [comment]
Updated version is up. The main change was to add support for more top-level keys in the default `install-config.yaml` template, and update the template vars list in the README accordingly.

**Comment by tehsmyers** [comment]
I don't know why travis hasn't tested this, but the reason it was failing was a py3 incompatibility in the unit tests that I (have probably) fixed, so...I'mma hit the big green button now.

**Pull request merged**